apps/leo_gateway/src/leo_gateway_s3_api.erl

```diff
@@ -603,44 +632,53 @@ put_object(Directive, Req, Key, #req_params{handler = ?PROTO_HANDLER_S3,
+                            put_object_1(Directive, Req, CS2, Key, Meta, RespObject, Params#req_params{custom_metadata = CMetaBin});
                         _TotalChunkedObjs ->
                             put_large_object_1(Directive, Req, Key, Meta, Params#req_params{custom_metadata = CMetaBin})
                     end;
                 {error, not_found} ->
+                    ?access_log_copy(BucketName, SrcDst, 0, ?HTTP_ST_NOT_FOUND, BeginTime),
                     ?reply_not_found([?SERVER_HEADER], Key, <<>>, Req);
                 {error, unavailable} ->
+                    ?access_log_copy(BucketName, SrcDst, 0, ?HTTP_ST_SERVICE_UNAVAILABLE, BeginTime),
                     ?reply_service_unavailable_error([?SERVER_HEADER], Key, <<>>, Req);
                 {error, ?ERR_TYPE_INTERNAL_ERROR} ->
+                    ?access_log_copy(BucketName, SrcDst, 0, ?HTTP_ST_INTERNAL_ERROR, BeginTime),
                     ?reply_internal_error([?SERVER_HEADER], Key, <<>>, Req);
                 {error, timeout} ->
+                    ?access_log_copy(BucketName, SrcDst, 0, ?HTTP_ST_SERVICE_UNAVAILABLE, BeginTime),
                     ?reply_timeout([?SERVER_HEADER], Key, <<>>, Req)
             end
     end.
 
 %% @doc POST/PUT operation on Objects. COPY
 %% @private
-put_object_1(Directive, Req, Key, Meta, Bin, #req_params{bucket_name = BucketName,
-                                                         bucket_info = BucketInfo,
-                                                         custom_metadata = CMetaBin} = Params) ->
-    BeginTime = leo_date:clock(),
+put_object_1(Directive, Req, Src, Key, Meta, Bin, #req_params{bucket_name = BucketName,
+                                                              bucket_info = BucketInfo,
+                                                              custom_metadata = CMetaBin,
+                                                              begin_time = BeginTime} = Params) ->
     Size = size(Bin),
+    SrcDst = <<Src/binary, " -> ", Key/binary>>,
     case leo_gateway_rpc_handler:put(#put_req_params{path = Key,
                                                      body = Bin,
                                                      meta = CMetaBin,
                                                      dsize = Size,
                                                      msize = byte_size(CMetaBin),
                                                      bucket_info = BucketInfo}) of
         {ok, _ETag} when Directive == ?HTTP_HEAD_X_AMZ_META_DIRECTIVE_COPY ->
-            ?access_log_put(BucketName, Key, Size, ?HTTP_ST_OK, BeginTime),
+            ?access_log_copy(BucketName, SrcDst, Size, ?HTTP_ST_OK, BeginTime),
             resp_copy_obj_xml(Req, Meta);
         {ok, _ETag} when Directive == ?HTTP_HEAD_X_AMZ_META_DIRECTIVE_REPLACE ->
+            ?access_log_copy(BucketName, SrcDst, Size, ?HTTP_ST_OK, BeginTime),
```

Comment:
@mocchira Actually here I don't understand why there would be a delete after the copy when `x-amz-metadata-directive` is set to `REPLACE`